worlds/witness/items.py

```diff
@@ -176,9 +176,16 @@ def get_filler_items(self, quantity: int) -> Dict[str, int]:
 
         # Read trap configuration data.
         trap_weight = self._world.options.trap_percentage / 100
-        filler_weight = 1 - trap_weight
+        trap_items = {name: self._world.options.trap_weights.get(name, 0)
+                      for (name, data) in self.item_data.items() if data.definition.category is ItemCategory.TRAP
+                      if isinstance(data.definition, WeightedItemDefinition)}
```

Comment:
I believe this should be an `and` and not an `if`-chain
```suggestion
                      and isinstance(data.definition, WeightedItemDefinition)}
```